spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/source/SparkBatchQueryScan.java

```diff
@@ -74,60 +101,176 @@
       throw new IllegalArgumentException("Cannot only specify option end-snapshot-id to do incremental scan");
     }
 
-    // look for split behavior overrides in options
-    this.splitSize = Spark3Util.propertyAsLong(options, SparkReadOptions.SPLIT_SIZE, null);
-    this.splitLookback = Spark3Util.propertyAsInt(options, SparkReadOptions.LOOKBACK, null);
-    this.splitOpenFileCost = Spark3Util.propertyAsLong(options, SparkReadOptions.FILE_OPEN_COST, null);
+    this.splitSize = readConf.splitSizeOption();
+    this.splitLookback = readConf.splitLookbackOption();
+    this.splitOpenFileCost = readConf.splitOpenFileCostOption();
+    this.runtimeFilterExpressions = Lists.newArrayList();
   }
 
-  @Override
-  protected List<CombinedScanTask> tasks() {
-    if (tasks == null) {
-      TableScan scan = table()
-          .newScan()
-          .caseSensitive(caseSensitive())
-          .project(expectedSchema());
+  private TableScan scan() {
+    if (scan == null) {
+      this.scan = buildScan();
+    }
 
-      if (snapshotId != null) {
-        scan = scan.useSnapshot(snapshotId);
-      }
+    return scan;
```

Comment:
Nit: The changes in this PR would be a bit smaller by naming the instance variable `tableScan` and leaving the variable named `scan` in `buildScan`.

Reply:
This would also make it easier to spot places where the variable is accessed directly instead of through the lazy access method, `scan`.

Reply:
Done.